### PR TITLE
DOC-457, DOC-435 Rework streams index

### PR DESCRIPTION
**Pull request by diana-macrometa**
- Broke up index.md topic into smaller topics
- Moved content out of Quickstart into Index intro topic
- No major content changes to any topic except for index.md
- Made new Jira tickets to more thoroughly review the new topics

### CODE REVIEW

File: docs/streams/index.md [hunk 1]
```diff
@@ -3,6 +3,26 @@ sidebar_position: 1
 title: Streams
 ---
 
+Streams are a type of collection in GDN to capture data in motion. Messages are sent via streams by publishers to consumers who then do something with the message. Streams can be created with client SDKs (pyC8, jsC8), REST APIs, or the web console.
```
Comment (StoyanTc) [inline docs/streams/index.md]: The streams are flows of data in GDN. They are not collections and is not mandatory to keep any data.

Comment (diana-macrometa) [reply docs/streams/index.md]: Updated. How's this?

As for keeping data, are you referring to the Message Retention and Expiry section in messages.md?

Comment (StoyanTc) [reply docs/streams/index.md]: Yes, the retention by default is 0. We have explicitly changed it to 1h on GDN.
Apart from this there are 4 collection types in GDN and the streams cannot be the 5th one.
This sounds confusing.

**Comment by diana-macrometa** [comment]
Hey all, please review index.md.

**Pull request merged**